simulator/controller/controller.go

```diff
@@ -0,0 +1,173 @@
+package controller
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"golang.org/x/xerrors"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/informers"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/metadata/metadatainformer"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/controller-manager/pkg/clientbuilder"
+	"k8s.io/controller-manager/pkg/informerfactory"
+	"k8s.io/klog/v2"
+	controllermanageroptions "k8s.io/kubernetes/cmd/kube-controller-manager/app/options"
+	kubectrlmgrconfig "k8s.io/kubernetes/pkg/controller/apis/config"
+)
+
+const (
+	// ControllerStartJitter is the Jitter used when starting controller.
+	ControllerStartJitter = 1.0
+)
+
+type InitFunc func(controllerCtx ControllerContext) error
+
+// ControllerInitializersFunc is used to create a collection of initializers
+//  given the loopMode.
+//nolint:revive // Intentionally used the same name.
+type ControllerInitializersFunc func() (initializers map[string]InitFunc)
+
+var _ ControllerInitializersFunc = NewControllerInitializers
+
+func NewController(client clientset.Interface, c *restclient.Config) (func(), error) {
```

Comment:
This function create controllers and start them. `NewController` looks "it only initialize controllers, but not start them".
So, I guess it should be renamed like `StartControllers`.